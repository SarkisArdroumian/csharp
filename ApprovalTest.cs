using System;
using System.IO;
using System.Text;
using ApprovalTests;
using ApprovalTests.Combinations;
using ApprovalTests.Reporters;
using NUnit.Framework;

namespace csharp
{
    [UseReporter(typeof(DiffReporter))]
    [TestFixture]
    public class ApprovalTest
    {
        [Test]
        public void ThirtyDays()
        {
            StringBuilder fakeoutput = new StringBuilder();
            Console.SetOut(new StringWriter(fakeoutput));
            Console.SetIn(new StringReader("a\n"));

            Program.Main(new string[] { });
            var output = fakeoutput.ToString();

            Approvals.Verify(output);
        }

        [Test]
        public void updateQuality()
        {
            CombinationApprovals.VerifyAllCombinations
            (
                doUpdateQuality,
                new string[]
                {
                     "+5 Dexterity Vest",
                     "Aged Brie",
                     "Elixir of the Mongoose",
                     "Sulfuras, Hand of Ragnaros",
                     "Backstage passes to a TAFKAL80ETC concert",
                     "Conjured Mana Cake"
                },
                new int[] { -1, 1, 0, 6, 11 },
                new int[] { 0, 10, 50 }
            );
        }

        private string doUpdateQuality(string name, int sellIn, int quality)
        {
            Item[] items = new Item[] { new Item { Name = name, SellIn = sellIn, Quality = quality } };

            GildedRose app = new GildedRose(items);
            app.UpdateQuality();

            return app.Items[0].ToString();
        }
    }
}
