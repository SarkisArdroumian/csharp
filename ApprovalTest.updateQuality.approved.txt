[+5 Dexterity Vest,-1,0] => +5 Dexterity Vest, -2, 0
[+5 Dexterity Vest,-1,10] => +5 Dexterity Vest, -2, 8
[+5 Dexterity Vest,-1,50] => +5 Dexterity Vest, -2, 48
[+5 Dexterity Vest,1,0] => +5 Dexterity Vest, 0, 0
[+5 Dexterity Vest,1,10] => +5 Dexterity Vest, 0, 9
[+5 Dexterity Vest,1,50] => +5 Dexterity Vest, 0, 49
[+5 Dexterity Vest,0,0] => +5 Dexterity Vest, -1, 0
[+5 Dexterity Vest,0,10] => +5 Dexterity Vest, -1, 8
[+5 Dexterity Vest,0,50] => +5 Dexterity Vest, -1, 48
[+5 Dexterity Vest,6,0] => +5 Dexterity Vest, 5, 0
[+5 Dexterity Vest,6,10] => +5 Dexterity Vest, 5, 9
[+5 Dexterity Vest,6,50] => +5 Dexterity Vest, 5, 49
[+5 Dexterity Vest,11,0] => +5 Dexterity Vest, 10, 0
[+5 Dexterity Vest,11,10] => +5 Dexterity Vest, 10, 9
[+5 Dexterity Vest,11,50] => +5 Dexterity Vest, 10, 49
[Aged Brie,-1,0] => Aged Brie, -2, 2
[Aged Brie,-1,10] => Aged Brie, -2, 12
[Aged Brie,-1,50] => Aged Brie, -2, 50
[Aged Brie,1,0] => Aged Brie, 0, 1
[Aged Brie,1,10] => Aged Brie, 0, 11
[Aged Brie,1,50] => Aged Brie, 0, 50
[Aged Brie,0,0] => Aged Brie, -1, 2
[Aged Brie,0,10] => Aged Brie, -1, 12
[Aged Brie,0,50] => Aged Brie, -1, 50
[Aged Brie,6,0] => Aged Brie, 5, 1
[Aged Brie,6,10] => Aged Brie, 5, 11
[Aged Brie,6,50] => Aged Brie, 5, 50
[Aged Brie,11,0] => Aged Brie, 10, 1
[Aged Brie,11,10] => Aged Brie, 10, 11
[Aged Brie,11,50] => Aged Brie, 10, 50
[Elixir of the Mongoose,-1,0] => Elixir of the Mongoose, -2, 0
[Elixir of the Mongoose,-1,10] => Elixir of the Mongoose, -2, 8
[Elixir of the Mongoose,-1,50] => Elixir of the Mongoose, -2, 48
[Elixir of the Mongoose,1,0] => Elixir of the Mongoose, 0, 0
[Elixir of the Mongoose,1,10] => Elixir of the Mongoose, 0, 9
[Elixir of the Mongoose,1,50] => Elixir of the Mongoose, 0, 49
[Elixir of the Mongoose,0,0] => Elixir of the Mongoose, -1, 0
[Elixir of the Mongoose,0,10] => Elixir of the Mongoose, -1, 8
[Elixir of the Mongoose,0,50] => Elixir of the Mongoose, -1, 48
[Elixir of the Mongoose,6,0] => Elixir of the Mongoose, 5, 0
[Elixir of the Mongoose,6,10] => Elixir of the Mongoose, 5, 9
[Elixir of the Mongoose,6,50] => Elixir of the Mongoose, 5, 49
[Elixir of the Mongoose,11,0] => Elixir of the Mongoose, 10, 0
[Elixir of the Mongoose,11,10] => Elixir of the Mongoose, 10, 9
[Elixir of the Mongoose,11,50] => Elixir of the Mongoose, 10, 49
[Sulfuras, Hand of Ragnaros,-1,0] => Sulfuras, Hand of Ragnaros, -1, 0
[Sulfuras, Hand of Ragnaros,-1,10] => Sulfuras, Hand of Ragnaros, -1, 10
[Sulfuras, Hand of Ragnaros,-1,50] => Sulfuras, Hand of Ragnaros, -1, 50
[Sulfuras, Hand of Ragnaros,1,0] => Sulfuras, Hand of Ragnaros, 1, 0
[Sulfuras, Hand of Ragnaros,1,10] => Sulfuras, Hand of Ragnaros, 1, 10
[Sulfuras, Hand of Ragnaros,1,50] => Sulfuras, Hand of Ragnaros, 1, 50
[Sulfuras, Hand of Ragnaros,0,0] => Sulfuras, Hand of Ragnaros, 0, 0
[Sulfuras, Hand of Ragnaros,0,10] => Sulfuras, Hand of Ragnaros, 0, 10
[Sulfuras, Hand of Ragnaros,0,50] => Sulfuras, Hand of Ragnaros, 0, 50
[Sulfuras, Hand of Ragnaros,6,0] => Sulfuras, Hand of Ragnaros, 6, 0
[Sulfuras, Hand of Ragnaros,6,10] => Sulfuras, Hand of Ragnaros, 6, 10
[Sulfuras, Hand of Ragnaros,6,50] => Sulfuras, Hand of Ragnaros, 6, 50
[Sulfuras, Hand of Ragnaros,11,0] => Sulfuras, Hand of Ragnaros, 11, 0
[Sulfuras, Hand of Ragnaros,11,10] => Sulfuras, Hand of Ragnaros, 11, 10
[Sulfuras, Hand of Ragnaros,11,50] => Sulfuras, Hand of Ragnaros, 11, 50
[Backstage passes to a TAFKAL80ETC concert,-1,0] => Backstage passes to a TAFKAL80ETC concert, -2, 0
[Backstage passes to a TAFKAL80ETC concert,-1,10] => Backstage passes to a TAFKAL80ETC concert, -2, 0
[Backstage passes to a TAFKAL80ETC concert,-1,50] => Backstage passes to a TAFKAL80ETC concert, -2, 0
[Backstage passes to a TAFKAL80ETC concert,1,0] => Backstage passes to a TAFKAL80ETC concert, 0, 3
[Backstage passes to a TAFKAL80ETC concert,1,10] => Backstage passes to a TAFKAL80ETC concert, 0, 13
[Backstage passes to a TAFKAL80ETC concert,1,50] => Backstage passes to a TAFKAL80ETC concert, 0, 50
[Backstage passes to a TAFKAL80ETC concert,0,0] => Backstage passes to a TAFKAL80ETC concert, -1, 0
[Backstage passes to a TAFKAL80ETC concert,0,10] => Backstage passes to a TAFKAL80ETC concert, -1, 0
[Backstage passes to a TAFKAL80ETC concert,0,50] => Backstage passes to a TAFKAL80ETC concert, -1, 0
[Backstage passes to a TAFKAL80ETC concert,6,0] => Backstage passes to a TAFKAL80ETC concert, 5, 2
[Backstage passes to a TAFKAL80ETC concert,6,10] => Backstage passes to a TAFKAL80ETC concert, 5, 12
[Backstage passes to a TAFKAL80ETC concert,6,50] => Backstage passes to a TAFKAL80ETC concert, 5, 50
[Backstage passes to a TAFKAL80ETC concert,11,0] => Backstage passes to a TAFKAL80ETC concert, 10, 1
[Backstage passes to a TAFKAL80ETC concert,11,10] => Backstage passes to a TAFKAL80ETC concert, 10, 11
[Backstage passes to a TAFKAL80ETC concert,11,50] => Backstage passes to a TAFKAL80ETC concert, 10, 50
[Conjured Mana Cake,-1,0] => Conjured Mana Cake, -2, 0
[Conjured Mana Cake,-1,10] => Conjured Mana Cake, -2, 6
[Conjured Mana Cake,-1,50] => Conjured Mana Cake, -2, 46
[Conjured Mana Cake,1,0] => Conjured Mana Cake, 0, 0
[Conjured Mana Cake,1,10] => Conjured Mana Cake, 0, 8
[Conjured Mana Cake,1,50] => Conjured Mana Cake, 0, 48
[Conjured Mana Cake,0,0] => Conjured Mana Cake, -1, 0
[Conjured Mana Cake,0,10] => Conjured Mana Cake, -1, 6
[Conjured Mana Cake,0,50] => Conjured Mana Cake, -1, 46
[Conjured Mana Cake,6,0] => Conjured Mana Cake, 5, 0
[Conjured Mana Cake,6,10] => Conjured Mana Cake, 5, 8
[Conjured Mana Cake,6,50] => Conjured Mana Cake, 5, 48
[Conjured Mana Cake,11,0] => Conjured Mana Cake, 10, 0
[Conjured Mana Cake,11,10] => Conjured Mana Cake, 10, 8
[Conjured Mana Cake,11,50] => Conjured Mana Cake, 10, 48
